### PR TITLE
Fix CS errors and issues reported by static analyzers

### PR DESCRIPTION
**Pull request by ADmad**
The phpcs ignores can be removed once https://github.com/slevomat/coding-standard/issues/851 is addressed.

### CODE REVIEW

File: src/Console/HelperRegistry.php [hunk 1]
```diff
@@ -55,8 +55,7 @@ public function setIo(ConsoleIo $io): void
      * Part of the template method for Cake\Core\ObjectRegistry::load()
      *
      * @param string $class Partial classname to resolve.
-     * @return string|null Either the correct class name or null.
-     * @psalm-return class-string
+     * @return class-string|null Either the correct class name or null.
```
Comment (othercorey) [inline src/Console/HelperRegistry.php]: TypeResolver doesn't seem to handle `class-string` yet. Maybe I can add it in a PR.

File: src/Cache/CacheRegistry.php [hunk 2]
```diff
@@ -36,8 +36,7 @@ class CacheRegistry extends ObjectRegistry
      * Part of the template method for Cake\Core\ObjectRegistry::load()
      *
      * @param string $class Partial classname to resolve.
-     * @return string|null Either the correct classname or null.
-     * @psalm-return class-string
+     * @return class-string|null Either the correct classname or null.
```
Comment (dereuromark) [inline src/Cache/CacheRegistry.php]: We shouldnt do this yet.

**Comment by othercorey** [comment]
Opened PR for class-string support in TypeResolver so we don't have to rely on psalm-specific annotations.

https://github.com/phpDocumentor/TypeResolver/pull/90

**Comment by ADmad** [comment]
Both phpstan and psalm (and even phan) support `class-string`, so others should catch up.

**Comment by dereuromark** [comment]
Maybe not IDEs and other tools? For them it is still string.

**Pull request merged**